### ISSUE TITLE
Improve error handling

### ISSUE THREAD

**Issue by miyao-yusuke** [mynlp/jigg#12]
To increase usability, errors in invoking/running annotators should be handled properly, and helpful messages should be shown.  For example,
- Show an instruction to download the model file when a CCG model file is not found.
- Instruct the installation of mecab etc. when the necessary command is not found.
- Improve the message when requirements are not satisfied.
- Show a message when a path specification is wrong (necessary libraries are not found).
- Check the version of java and show a message if a wrong version is used.

**Comment by hiroshinoji** [comment]
Supported in #21.

For the last error (version error), I found that the problem is not that the source or library is compatible with Java 1.7, but that the current distributed jar is compiled with Java 1.7. When I downgrade my Java version to 1.6, assembly was successful, and it worked on Java 1.6.

More specifically, I found that the problem occurs due to a combination of sbt and javac. If I add additional arguments for javac option in build.sbt, like

``` Scala
javacOptions ++= Seq("-source", "1.6", "-target", "1.6")
```

, which is added in 23c0e7a1be4b7d39480b6dfa15dbf12a96c248d0, and then build the jar with Java 1.7 or Java 1.8, the resulting jar does not throw version error even with Java 1.6.

These options force the resulting jar to be compatible with the specified version. The problem occurs only due to java sources, not scala sources, so these options are perfect to create a binary to be compatible with Java 1.6. I also found that in the current setting, compile fails if we try to use functionalities of Java >= 1.7, which is ok.

Thus, if we release the latest jar with the current setting of sbt, that would be worked for users with Java 1.6.

When we use a library which does not support 1.6, such as Stanford CoreNLP, compile still succeeds, but an error is detected when trying to load classes of that library. I think this is ok, since it means a user need not to upgrade Java unless one finds that the annotator currently tried is not compatible with the current version of Java. Now some help message to promote upgrading Java is shown when such version error is detected at runtime.
